src/query/query_executor.c

```diff
@@ -20078,14 +20068,17 @@ qexec_analytic_eval_instnum_pred (THREAD_ENTRY * thread_p, ANALYTIC_STATE * anal
       return NO_ERROR;
     }
 
+  analytic_state->xasl->instnum_flag &= ~(XASL_INSTNUM_FLAG_SCAN_STOP_AT_ANALYTIC);
   /* evaluate inst_num() */
   is_output_rec = qexec_eval_instnum_pred (thread_p, analytic_state->xasl, analytic_state->xasl_state);
   if (is_output_rec == V_ERROR)
     {
+      analytic_state->xasl->instnum_flag |= XASL_INSTNUM_FLAG_SCAN_STOP_AT_ANALYTIC;
```

Comment:
XASL_INSTNUM_FLAG_SCAN_STOP_AT_ANALYTIC is always set. The flag XASL_INSTNUM_FLAG_SCAN_STOP_AT_ANALYTIC setting can be moved out of the if-else clause.

Reply:
can't be moved out of the if-else because of return.
I fixed reset of flag.